nexus/src/nexus.rs

```diff
@@ -209,28 +209,28 @@ impl Nexus {
         };
 
         /* TODO-cleanup all the extra Arcs here seems wrong */
-        let nexus_arc = Arc::new(nexus);
+        let nexus = Arc::new(nexus);
         let opctx = OpContext::for_background(
             log.new(o!("component" => "SagaRecoverer")),
             authz,
             authn::Context::internal_saga_recovery(),
             Arc::clone(&db_datastore),
         );
+        let saga_logger = nexus.log.new(o!("saga_type" => "recovery"));
```

Comment:
When creating the other logger in `execute_saga`, I created the logger with the key `template_name`.

In this recovery setup, however, we create a `SagaContext` object before we know what templates we'll be processing.

So basically, yeah: I wanted some way to distinguish the "saga context for recovery" vs "the normal sagas".

This is totally arbitrary though, if we'd prefer different keys, this could change.

Reply:
Sounds reasonable.  It'll be good to eventually get the template name in those too but we can do that later!